libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -96,6 +96,18 @@ void AP_AHRS_NavEKF::update(bool skip_ins_update)
         update_EKF3();
         update_EKF2();
     }
+
+    // update home location for GPS baro correction for DCM
+    Location EKF_home_loc;
+    if (_ekf_type == 2 && _ekf2_started && (EKF2.getOriginHgtMode() & (1<<2)) && EKF2.healthy()) {
+        EKF2.getOriginLLH(-1, EKF_home_loc);
```

Comment:
Yes, I am worried about a jump in height if the AHRS falls back to DCM for plane.
If it is needed to change the way it is handled, I will do, however this height jump may be very fatal if happens at low heights.

As you wrote in previous PR:
> unless AP_AHRS_DCM::get_relative_position_D_home() is modified to correct using the difference between the HOME altitude and last known good EKF origin altitude. Bit position 2 in the _OGN_HGT_MASK parameter would need to be set to 2 so that the drift correction was applied to the EKF origin for this to work.